MonoGame.Framework/Platform/Utilities/FuncLoader.Desktop.cs

```diff
@@ -43,6 +43,8 @@ public static IntPtr LoadLibraryExt(string libname)
             // Try .NET Framework / mono locations
             if (CurrentPlatform.OS == OS.MacOSX)
             {
+                // Fix assembly location for macOS .app bundles
+                assemblyLocation = System.AppDomain.CurrentDomain.BaseDirectory;
```

Comment:
You shouldn't be modifying assemblyLocation, it will affect a lot of other ways of packaging stuff. Instead you would need to add one more if ret == IntPtr.Zero check.

Also noticed that you are using `Dotnet.Bundle` package, however as that thing is not open source, I don't think we should even support it.

Reply:
https://github.com/egramtel/dotnet-bundle this should be  the source of `Dotnet.Bundle`.
I folllowed this doc https://avaloniaui.net/docs/packing/macOS and I'm just too lazy to write the `Info.plist` for app bundle so I choose to use it to pack a app bundle.

And I believe it's not about `Dotnet.Bundle`, what it does is simply help us write `Info.plist` and copy files. I tried move *.dylib files to `.app/Content/Frameworks` directory and it's the same error. 

I'd print the `Directory.GetCurrentDirectory()` and it shows the current working directory is `/`, and then print  `assemblyLocation` is `./`, that's why search *.dylib fails.

Reply:
What I'm trying to tell you is to do the following:
```cs
// Look in .app bundle
if (ret == IntPtr.Zero)	
    ret = LoadLibrary(Path.Combine(Directory.GetCurrentDirectory(), "something something"));
```

`assemblyLocation` is used outside of the if statement for a lot of other packaging methods, you should not modify it.

Reply:
I just pushed and dont' change `assemblyLocation `.